src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -117,6 +117,13 @@ public void configurePipeline(PipelineConfigurer configurer) {
   public void prepareRun(BatchSourceContext context) throws Exception {
     FailureCollector collector = context.getFailureCollector();
     config.validate(collector);
+
+    if (getBQSchema(collector).getFields().isEmpty()) {
+      collector.addFailure("No schema was specified in input BigQuery table.",
```

Comment:
better to include the table name:

BigQuery table '[dataset].[table]' does not have a schema. Please edit the table to add a schema.